kie-server-parent/kie-server-services/kie-server-services-kafka/src/test/java/org/kie/server/services/jbpm/kafka/KakfaServerExtensionTest.java

```diff
@@ -0,0 +1,149 @@
+import org.kie.server.services.impl.KieServerImpl;
+import org.mockito.Mockito;
+import org.mockito.verification.VerificationMode;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+public class KakfaServerExtensionTest {
+
+    private static class MockKafkaServerExtension extends KafkaServerExtension {
+
+        private MockConsumer<String, byte[]> consumer = new MockConsumer<String, byte[]>(OffsetResetStrategy.EARLIEST);
+
+        protected Consumer<String, byte[]> getKafkaConsumer() {
+            return consumer;
+        }
+
+        public MockConsumer<String, byte[]> getMockConsumer() {
+            return consumer;
+        }
+    }
+
+    private ProcessService processService;
+    private MockKafkaServerExtension extension;
+    private KieServerImpl server;
+    private KieServerRegistry registry;
+    private DeployedUnit deployedUnit;
+    private ProcessDefinition processDefinition;
+
+    @Before
+    public void setup() {
+        extension = new MockKafkaServerExtension();
+        server = mock(KieServerImpl.class);
+        registry = mock(KieServerRegistry.class);
+        KieServerExtension serverExtension = mock(KieServerExtension.class);
+
+        when(registry.getServerExtension(Mockito.anyString())).thenReturn(serverExtension);
+        ListenerSupport deployService = mock(ListenerSupport.class, withSettings().extraInterfaces(
+                DeploymentService.class));
+        processService = mock(ProcessService.class);
+        when(serverExtension.getServices()).thenReturn(Arrays.asList(deployService, processService));
+        extension.init(server, registry);
+        extension.serverStarted();
+        deployedUnit = mock(DeployedUnit.class);
+        processDefinition = mock(ProcessDefinition.class);
+        when(deployedUnit.getDeployedAssets()).thenReturn(Collections.singletonList(processDefinition));
+    }
+
+    @After
+    public void close() {
+        extension.destroy(server, registry);
+    }
+
+    @Test
+    public void testKafkaServerExecutorSignal() throws InterruptedException {
+        when(processDefinition.getSignalsMetadata()).thenReturn(Collections.singletonList(SignalDescImpl.from(
+                new Signal("MySignal", "MySignal", "String"))));
+        when(deployedUnit.getDeployedAssets()).thenReturn(Collections.singletonList(processDefinition));
+        extension.onDeploy(new DeploymentEvent("MyDeploy", deployedUnit));
+        publishEvent("MySignal", "{\"id\":\"javi\",\"type\":\"one\",\"source\":\"pepe\",\"data\":\"javierito\"}");
+        verify(processService, getTimeout()).signalEvent("MyDeploy", "MySignal", "javierito");
+    }
+
+    @Test
+    public void testKafkaServerExecutorMessage() throws InterruptedException {
```

Comment:
Checked exception is not thrown within this method
```suggestion
    public void testKafkaServerExecutorMessage() {
```